Extension/CloudFoundryCLI/cloudfoundrycli.ts

```diff
@@ -13,11 +15,24 @@ var cfEndpointUrl = tl.getEndpointUrl(cfEndpoint, false);
 var cfEndpointAuth = tl.getEndpointAuthorization(cfEndpoint, false);
 var workingDir = tl.getInput('workingDirectory', true);
 var cfPath = tl.which('cf');
+
+
 if (tl.filePathSupplied('cfToolLocation')) {
     tl.debug('Using supplied tool location');
     cfPath = tl.getPathInput('cfToolLocation');
 }
 
+function getOptions() {
+    const tempDir = tl.getVariable('Agent.TempDirectory');
+
+    var options = <IExecOptions>{};
+    options.env = process.env;
+    // overriding config dir so that different agents on the same host can run. https://docs.cloudfoundry.org/cf-cli/cf-help.html 
+    options.env['CF_HOME'] = path.join(tempDir, 'cfCLI');
```

Comment:
Hi @mozts2005!  It seems that this is equivalent to setting the `CF_HOME` variable in the pipeline settings to `$(Agent.TempDirectory)`.  Is there more context to this change that I might be missing?

Perhaps you have more insight on this: how common is it to override the `CF_HOME` setting?  I'm worried that this change might affect every other user who has this set to something else in their pipelines.

Thanks!

Reply:
@lkillgore, the point of this change is to set a default that allows multiple agents to run on the same server.

Reply:
Hi @mozts2005, thanks for your response, and I apologize if I'm not explaining myself well.

As far as I can tell, (and please correct me where I am wrong) a user could get the same results with the existing task if they set a variable for the pipeline: `CF_HOME: $(Agent.TempDirectory)/cfCLI`.

For existing users who have no need for this change, it seems to me that this change could have side effects that are undesirable.  For instance, if a user has a build/pipeline and has already set `CF_HOME` to another value, it seems that this could change their results and possibly break them.

Reply:
Based on my testing If I set the "CF_home" on the pipeline, it is never passed to the cf cli. But if you would like, I can update this to allow an override to be set if they do add it as a pipeline variable.

Reply:
Hrmm... that's curious.  Based on my reading of the code, I would expect that it _should_ work.  I wonder if there's a  separate bug in the mix.  How have you been testing the value for CF_HOME with the CLI?

Reply:
I am going to update this to default to the agent-based path but allow override by the user.

Reply:
Thanks @mozts2005!

After discussing this change with @leantk, this is what we're thinking: we'd like this change to be an opt-in feature.  Adding a checkbox seems to be the easiest way to do that.  Maybe something like `Use agent-specific 'CF_HOME'`.  Just in case a user is implicitly relying on the default value for `CF_HOME`.  (I'm not an expert on how this variable is overridden, so perhaps this is a bit silly.  But, I'd hate to needlessly break existing pipelines when a user has their task updated.)

I'm happy to accept this PR as-is, and make the change myself.  Or, you can add the checkbox yourself.  Please let me know what you'd like to do.  (For an example of the checkbox UI, take a look at [this](https://github.com/microsoft/azure-pipelines-tasks/blob/4795bce4b292080f90ce1120ed4620f652403859/Tasks/MavenV3/task.json#L79).)

Reply:
I can make that change but to be clear NO ever changes CF_Home it was only added to allow Pivotal staff to have more then one clients creds this is only used for storing settings it is not used as work directory.

Reply:
If that's the case, I think I'd prefer to take this change as-is.

Reply:
After discussing with @leantk, we'll take this as-is.  I'm going to merge your change, and then do some of the housekeeping necessary to release this in the marketplace.